tools/SampleAnalyzer/Commons/DataFormat/MCSampleFormat.h

```diff
@@ -63,182 +59,165 @@ namespace MA5
         friend class STDHEPReader;
         friend class STDHEPreader;
 
-
         // -------------------------------------------------------------
         //                        data members
         // -------------------------------------------------------------
     private:
-
         // ---------------------- physics info -------------------------
-        std::pair<MAint32,MAint32>     beamPDGID_;
-        std::pair<MAfloat64,MAfloat64> beamE_;
-        std::pair<MAuint32,MAuint32>   beamPDFauthor_;
-        std::pair<MAuint32,MAuint32>   beamPDFID_;
-        MAint32                        weightMode_;
-        std::vector<ProcessFormat>     processes_;
-        const MA5GEN::GeneratorType*   sample_generator_;
-        MAfloat32                      length_unit_; /// Length unit: mm=1 cm=0.1
-        MAfloat32                      energy_unit_; /// Energy unit: GeV=1 MeV=0.001 keV=0.000001
+        std::pair<MAint32, MAint32> beamPDGID_;
+        std::pair<MAfloat64, MAfloat64> beamE_;
+        std::pair<MAuint32, MAuint32> beamPDFauthor_;
+        std::pair<MAuint32, MAuint32> beamPDFID_;
+        MAint32 weightMode_;
```

Comment:
@BFuks What is weight mode?

Reply:
That's something related to the weighting strategy to be used in LHE files. That comes from the early days.